yt_dlp/extractor/tiktok.py

```diff
@@ -593,7 +626,10 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, user_name, headers={
             'User-Agent': 'facebookexternalhit/1.1 (+http://www.facebook.com/externalhit_uatext.php)'
         })
-        user_id = self._html_search_regex(r'snssdk\d*://user/profile/(\d+)', webpage, 'user ID')
+        user_id = self._html_search_regex(r'snssdk\d*://user/profile/(\d+)', webpage, 'user ID', default=None)
+        if not user_id:
+            user_id = self._search_regex(r'\<meta\s+property=\"og:url\"\s+content=\"https?.+?\@(?P<user_id>\d+)\W',
+                                         webpage, 'user ID')
```

Comment:
Don't unnecessarily escape chars in regex

Reply:
In here we have 2 case `https://tiktok.com/@<user_name>` and `https://tiktok.com/@<user_id>`, and for the second case, the regex above cant find the user_id. so i use this regex to get the user_id, but come to think of it, I don't think we need it and we can set user_id = user_name